source/common/stream_info/filter_state_impl.cc

```diff
@@ -25,9 +25,19 @@ void FilterStateImpl::setData(absl::string_view data_name, std::unique_ptr<Objec
   std::unique_ptr<FilterStateImpl::FilterObject> filter_object(new FilterStateImpl::FilterObject());
   filter_object->data_ = std::move(data);
   filter_object->state_type_ = state_type;
+  filter_object->life_span_ = life_span;
```

Comment:
I may have missed something, but filter_object still needs to be a unique_ptr because data_storage_ stores unique_ptr. The filter_object->data_ now is shared_ptr. I removed the std::move on data.